lib/msf/core/exploit/remote/http/http_cookie.rb

```diff
@@ -0,0 +1,195 @@
+          include Comparable
+
+          def initialize(name, value = nil, **attr_hash)
+            if name.is_a?(::HTTP::Cookie)
+              @cookie = name
+            elsif name && value && attr_hash
+              @cookie = ::HTTP::Cookie.new(name, value, **attr_hash)
+            elsif name && value
+              @cookie = ::HTTP::Cookie.new(name, value)
+            else
+              @cookie = ::HTTP::Cookie.new(name)
+            end
+          end
+
+          def name
+            @cookie.name
+          end
+
+          def name=(name)
+            @cookie.name = name.to_s
+          end
+
+          def value
+            @cookie.value
+          end
+
+          def value=(value)
+            if value.nil? || value.is_a?(String)
+              @cookie.value = value
+            else
+              @cookie.value = value.to_s
+            end
+          end
+
+          def max_age
+            @cookie.max_age
+          end
+
+          def max_age=(max_age)
+            if max_age.nil? || max_age.is_a?(Integer)
+              @cookie.max_age = max_age
+            else
+              @cookie.max_age = Integer(max_age)
+            end
+          end
+
+          def expires
+            @cookie.expires
+          end
+
+          def expires=(expires)
+            if expires.nil? || expires.is_a?(Time)
+              @cookie.expires = expires
+            else
+              t = Time.parse(expires)
+              @cookie.expires = t
+            end
+          end
+
+          def expired?(time = Time.now)
+            @cookie.expired?(time)
+          end
+
+          def path
+            @cookie.path
+          end
+
+          def path=(path)
+            if path.nil? || path.is_a?(String)
+              @cookie.path = path
+            else
+              @cookie.path = path.to_s
+            end
+          end
+
+          def secure
+            @cookie.secure
+          end
+
+          def secure=(secure)
+            @cookie.secure = !!secure
+          end
+
+          def httponly
+            @cookie.httponly
+          end
+
+          def httponly=(httponly)
+            @cookie.httponly = !!httponly
+          end
+
+          # def for_domain
+          #   @cookie.for_domain
+          # end
+          #
+          # def for_domain= for_domain
+          #   @cookie.for_domain= !!for_domain
+          # end
+
+          # def origin
+          #   @cookie.origin
+          # end
+          #
+          # def origin= origin
+          #   if origin.kind_of?(URI)
+          #     @cookie.origin= origin
+          #   else
+          #     @cookie.origin= origin.to_s
+          #   end
+          # end
+
+          def domain
+            @cookie.domain
+          end
+
+          def domain=(domain)
+            if domain.nil? || domain.is_a?(DomainName)
+              @cookie.domain = domain
+            else
+              @cookie.domain = DomainName(domain.to_s)
+            end
+          end
```

Comment:
Looks like this code goes through the same `DomainName` handling logic behind the scenes still, is that desired? 🤔

Reply:
Replaced `@cookie.domain = DomainName(domain.to_s)` with `@cookie.domain = domain.to_s` to ensure we don't lose purposely poorly formed url to the `DomainName` conversion